### PR TITLE
feat: webdriver7

### PR DESCRIPTION
**Pull request by jasonschroeder-sfdc**
This change does the following:

- Upgrade `webdriverio` packages and services to v`7`
- Added Node.js 16 to the test matrix
- Disabled eslint in a few places, as Jest27 cannot yet declare custom test matchers until Jest 28 (ref: https://jestjs.io/blog/2022/04/25/jest-28#expect)

This change is needed as:
- Node.js 14 is in Maintenance mode and will be EOL in April 2023 (ref: https://github.com/nodejs/release#release-schedule
- Webdriverio 6 LTS ended in Feb 2022 (ref: https://webdriver.io/versions)

### CODE REVIEW

File: packages/wdio/src/wdio.ts [hunk 1]
```diff
@@ -96,6 +96,7 @@ export async function assertAccessible(opts: Partial<WdioOptions> = {}): Promise
  * Verify that the currently loaded page in the browser is accessible.
  * Throw an error with the accessibility issues found if it is not accessible.
  * Synchronous version of {@link assertAccessible}
+ * @deprecated Please update to using async method.
```
Comment (mohanraj-r) [inline packages/wdio/src/wdio.ts:99]: 👍 The sync vs async WDIO APIs were a mess. Thanks for cleaning it up.

**Review by mohanraj-r — APPROVED** [review]
LGTM

**Comment by callenmas14** [comment]
I'm going by @mohanraj-r 's comment of LGTM.

**Review by callenmas14 — APPROVED** [review]
There's a lot I don't understand here.  I'm going by @mohanraj-r 's LGTM comment :) .

**Pull request merged**